python/tvm/octo/utils/target_info.py

```diff
@@ -0,0 +1,156 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=invalid-name, wrong-import-position
+"""Utility functions for finding information about current device."""
+import os
+import re
+import sys
+import psutil
+import subprocess
+import shutil
+import tvm
+
+
+def get_llvm_target() -> tvm.target.Target:
+    """Extract fully featured llvm target for current device.
+
+    Returns
+    -------
+    target : tvm.target.Target
+        A TVM target that fully describes the current devices CPU.
+    """
+    # If we cant find llc, we wont be able to extract more information.
+    if shutil.which("llc") is None:
+        return "llvm"
+
+    # Get host information from llc
+    stream = subprocess.run(["llc", "--version"], stdout=subprocess.PIPE)
+    cpu_info = stream.stdout.decode("utf-8")
+
+    # Parse out cpu line
+    cpu = re.search("(?<=Host CPU: ).+", cpu_info).group(0)
+
+    # Next extract attribute string.
+    platform = sys.platform
+    # Linux
+    if platform in ["linux", "linux2"]:
+        output = subprocess.check_output("lscpu", shell=True).decode()
+        pattern = r"^([^:]+):\s+(.*)$"
+        cpu_info = {}
+
+        for line in output.splitlines():
+            match = re.match(pattern, line)
+            if match:
+                key = match.group(1)
+                value = match.group(2)
+                cpu_info[key] = value.lower().strip()
+
+        features = cpu_info["Flags"].split(" ")
+        march = cpu_info["Architecture"]
+        cores = cpu_info["Core(s) per socket"]
+        sockets = cpu_info["Socket(s)"]
+        total_cores = str(int(cores) * int(sockets))
+        # Special case for x86_64 mismatch between underscore and hyphen
+        if march == "x86_64":
+            march = "x86-64"
+    else:
+        raise ValueError("Platform %s is not supported." % platform)
+
+    # Now we'll extract the architecture of the target.
+    output = subprocess.check_output("llc --version", shell=True).decode()
+    # Remove header.
+    march_options = re.search("(?<=Registered Targets:).*", output, re.DOTALL).group(0)
+    march_list = []
```

Comment:
small nit, could collapse to `march_list = [m.strip.split(" ") for m in march_options.split("\n") if m]`